app/script/util/DebugUtil.js

```diff
@@ -58,6 +58,44 @@ z.util.DebugUtil = class DebugUtil {
       .then(() => this.logger.log(`Corrupted Session ID '${sessionId}'`));
   }
 
+  haveISentThisMessageToMyOtherClients(
+    messageId,
+    conversationId = wire.app.repository.conversation.active_conversation().id
+  ) {
+    let amountOfMessagesSent = 0;
+    let dateTime = undefined;
+
+    const clientId = wire.app.repository.client.currentClient().id;
+    const userId = wire.app.repository.user.self().id;
+
+    const isOTRMessage = notification => notification.type === z.event.Backend.CONVERSATION.OTR_MESSAGE_ADD;
+    const isInCurrentConversation = notification => notification.conversation === conversationId;
+    const wasSentByOurCurrentClient = notification =>
+      notification.from === userId && (notification.data && notification.data.sender === clientId);
+    const hasExpectedTimestamp = notification => notification.time === dateTime.toISOString();
```

Comment:
I see temporal coupling here with the `dateTime` variable.
If, for some reason, latter we move the order of the `then` callbacks or someone overwrite the `dateTime` variable, things will break (silently or not).